### PR TITLE
feat: `createSpotifyOAuth2Client()`

### PR DESCRIPTION
**Pull request by Jabolol**
This PR implements the [Spotify Oauth2](https://developer.spotify.com/documentation/web-api/tutorials/code-flow) client. Updated `README.md`, added `Spotify` to the tests & demo, created Spotify's `client` config.

### CODE REVIEW

File: src/create_oauth2_client_test.ts [hunk 1]
```diff
@@ -13,6 +13,7 @@ import * as createOAuth2ClientFns from "./create_oauth2_client.ts";
   "Patreon",
   "Slack",
   "Twitter",
+  "Spotify",
```
Comment (iuioiua) [inline src/create_oauth2_client_test.ts:16]:
```suggestion
  "Slack",
  "Spotify",
  "Twitter",
```

**Review by iuioiua — CHANGES_REQUESTED** [review]
Mostly LGTM! Though, can you please take a look at the new [Adding a Pre-configured OAuth 2.0 Client README section](https://github.com/denoland/deno_kv_oauth#adding-a-pre-configured-oauth-20-client)? It details what we should include in these kinds of PRs from now on.

Please let me know if that new section doesn't answer all questions surrounding these PRs.

**Comment by Jabolol** [comment]
Everything is clear in the `README` and the snippet does work. Here's the screenshot, thanks!

image

**Review by iuioiua — APPROVED** [review]
LGTM! Thanks again 🙏🏾

**Pull request merged**